graylog2-web-interface/src/components/indices/IndexMaintenanceStrategiesConfiguration.jsx

```diff
@@ -0,0 +1,115 @@
+    description: React.PropTypes.string.isRequired,
+    selectPlaceholder: React.PropTypes.string.isRequired,
+    pluginExports: React.PropTypes.array.isRequired,
+    strategies: React.PropTypes.array.isRequired,
+    activeConfig: React.PropTypes.object.isRequired,
+    updateState: React.PropTypes.func.isRequired,
+  },
+
+  getInitialState() {
+    return {
+      activeStrategy: this.props.activeConfig.strategy,
+      activeConfig: this.props.activeConfig.config,
+      newStrategy: this.props.activeConfig.strategy,
+      newConfig: this.props.activeConfig.config,
+    };
+  },
+
+  _getDefaultStrategyConfig(selectedStrategy) {
+    const result = this.props.strategies.find((strategy) => strategy.type === selectedStrategy);
+    return result ? result.default_config : undefined;
+  },
+
+  _getStrategyJsonSchema(selectedStrategy) {
+    const result = this.props.strategies.find((strategy) => strategy.type === selectedStrategy);
+    return result ? result.json_schema : undefined;
+  },
+
+  _getStrategyConfig(selectedStrategy) {
+    if (this.state.activeStrategy === selectedStrategy) {
+      // If the newly selected strategy is the current active strategy, we use the active configuration.
+      return this.state.activeConfig;
+    } else {
+      // If the newly selected strategy is not the current active strategy, we use the selected strategy's default config.
+      return this._getDefaultStrategyConfig(selectedStrategy);
+    }
+  },
+
+  _onSelect(newStrategy) {
+    if (!newStrategy || newStrategy.length < 1) {
+      this.setState({newStrategy: undefined});
+      return;
+    }
+
+    const newConfig = this._getStrategyConfig(newStrategy);
+
+    this.setState({newStrategy: newStrategy, newConfig: newConfig});
+    this.props.updateState(newStrategy, newConfig);
+  },
+
+  _onConfigUpdate(newConfig) {
+    this.setState({newConfig: newConfig});
+    this.props.updateState(this.state.newStrategy, newConfig);
+  },
+
+  _availableSelectOptions() {
+    return this.props.pluginExports.map((config) => {
+      return {value: config.type, label: config.displayName}
+    });
+  },
+
+  _getConfigurationComponent(selectedStrategy) {
+    if (!selectedStrategy || selectedStrategy.length < 1) {
+      return;
+    }
+
+    const strategy = this.props.pluginExports.find((strategy) => strategy.type === selectedStrategy);
+
+    if (!strategy) {
+      return undefined;
+    }
+
+    const strategyConfig = this._getStrategyConfig(selectedStrategy);
+    const element = React.createElement(strategy.configComponent, {
+      config: strategyConfig,
+      jsonSchema: this._getStrategyJsonSchema(selectedStrategy),
+      updateConfig: this._onConfigUpdate
+    });
+
+    return (<span key={strategy.type}>{element}</span>);
+  },
+
+  _activeSelection() {
+    return this.state.newStrategy;
+  },
+
+  render() {
+    return (
+      <span>
+        <h3>{this.props.title}</h3>
+        <div style={{marginTop: 10}}>
```

Comment:
Is there a need for inline style definitions? I know we are using them a lot in the existing codebase, but this is mainly because of legacy HTML that was ported over. For new code we should extract those into separate CSS/LESS fragments.